### PR TITLE
Add looker example

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**
A small example which looks at the closest entity.

### CODE REVIEW

File: examples/looker.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+if (process.argv.length < 4 || process.argv.length > 6) {
+  console.log('Usage : node echo.js <host> <port> [<name>] [<password>]')
+  process.exit(1)
+}
+
+const bot = mineflayer.createBot({
+  host: process.argv[2],
+  port: parseInt(process.argv[3]),
+  username: process.argv[4] ? process.argv[4] : 'totem',
```
Comment (Karang) [inline examples/looker.js:15]: name it 'looker' ?

File: examples/looker.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  password: process.argv[5]
+})
+
+bot.once('spawn', function () {
+  setInterval(() => {
+    var entity = nearestEntity()
+    if (entity) {
+      if (entity.type === 'player') {
+        bot.lookAt(entity.position.offset(0, 1, 0))
```
Comment (Karang) [inline examples/looker.js:24]: If you want to look in the player eye, you can add 1.6 (the player eye height)

File: examples/looker.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        bot.lookAt(entity.position)
+      }
+    }
+  }, 50)
+})
+
+function nearestEntity (type) {
+  var id, entity, dist
+  var best = null
+  var bestDistance = null
```
Comment (wvffle) [inline examples/looker.js:35]: Use const/let

File: examples/looker.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    entity = bot.entities[id]
+    if (type && entity.type !== type) continue
+    if (entity === bot.entity) continue
+    dist = bot.entity.position.distanceTo(entity.position)
+    if (!best || dist < bestDistance) {
+      best = entity
+      bestDistance = dist
+    }
+  }
+  return best
```
Comment (wvffle) [inline examples/looker.js:46]: separate with newlines for readability

**Comment by AlexProgrammerDE** [comment]
oops forgot that i have opened this. I will adress your requests now.

File: examples/looker.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  let bestDistance = null
+
+  for (id in bot.entities) {
+    entity = bot.entities[id]
+
+    if (type && entity.type !== type) continue
+
+    if (entity === bot.entity) continue
+
+    dist = bot.entity.position.distanceTo(entity.position)
```
Comment (wvffle) [inline examples/looker.js:44]: this could be const

File: examples/looker.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  }, 50)
+})
+
+function nearestEntity (type) {
+  let id, entity, dist
+  let best = null
+  let bestDistance = null
+
+  for (id in bot.entities) {
```
Comment (wvffle) [inline examples/looker.js:37]: id can be const

File: examples/looker.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    entity = bot.entities[id]
```
Comment (wvffle) [inline examples/looker.js:38]: this can be const

**Comment by AlexProgrammerDE** [comment]
@wvffle i don't understand why declaring the type of the variable is so important. It works fine without it.

**Comment by rom1504** [comment]
it's fine

just FYI javascript is a very unsafe language. To avoid one kind of error, it's good to use const to avoid a variable that is supposed not to change to be reassigned.

**Comment by wvffle** [comment]
It's an **example**. Many new javascript devs  will look at it and use it as a source of their knowledge. We should teach good patterns, not bad ones. Having a global variable or a variable that is in the upper scope is a most likely a memleak. 

Also, changing the value type of a variable is really slow. (idk if it's slow with `undefined` but I doubt, dunno what with `null`)

File: examples/looker.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+const bot = mineflayer.createBot({
+  host: process.argv[2],
+  port: parseInt(process.argv[3]),
+  username: process.argv[4] ? process.argv[4] : 'looker',
+  password: process.argv[5]
+})
+
+bot.once('spawn', function () {
+  setInterval(() => {
+    var entity = nearestEntity()
```
Comment (wvffle) [inline examples/looker.js:21]: There is also a `var` which is pretty much unsafe. You should use `const` there.

**Comment by wvffle** [comment]
I'm for reverting this and fixing this issues.

**Comment by AlexProgrammerDE** [comment]
ok do it if you want.

**Comment by rom1504** [comment]
@wvffle fixed there https://github.com/PrismarineJS/mineflayer/pull/1147

**Pull request merged**